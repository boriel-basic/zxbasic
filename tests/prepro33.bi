asm
   EX DE,HL
   PUSH DE ; save our address

   LD L,(IX+8) ; data address
   LD H,(IX+9)
   EX AF,AF'
   LD A,4   ;row counter
   EX AF,AF'
   BLPutCharLoop:
   LD B,8
  
   BLPutCharOneCharLoop:
   XOR A; LD A,0
   LD C,A
   
   ; gets screen address in DE, and bytes address in HL. Copies the row to the screen
   LDI  ; also decrements B.
   LDI
   LDI
   LDI ; 4 bytes copied.
   LD A,C
   ADD A,E ;(A should be -4)
   LD E,A
   INC D
   LD A,B
   OR A
   JP NZ,BLPutCharOneCharLoop
   
   EX AF,AF'
   DEC A
   JR Z, BLPutCharsEnd  ; We've done all 4 rows.
   EX AF,AF'
end asm

