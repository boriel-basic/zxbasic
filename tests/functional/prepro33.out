#line 1 "prepro33.bi"
asm
   EX DE,HL
   PUSH DE 

   LD L,(IX+8) 
   LD H,(IX+9)
   EX AF,AF'
   LD A,4   
   EX AF,AF'
   BLPutCharLoop:
   LD B,8
  
   BLPutCharOneCharLoop:
   XOR A
   LD C,A
   
   
   LDI  
   LDI
   LDI
   LDI 
   LD A,C
   ADD A,E 
   LD E,A
   INC D
   LD A,B
   OR A
   JP NZ,BLPutCharOneCharLoop
   
   EX AF,AF'
   DEC A
   JR Z, BLPutCharsEnd  
   EX AF,AF'
end asm

